ci/images/plugin-compiler/data/build.sh

```diff
@@ -1,15 +1,32 @@
 #!/bin/bash
-set -xe
+set -e
+
+GATEWAY_VERSION=$(perl -n -e'/v(\d+).(\d+).(\d+)/'' && print "v$1\.$2\.$3"' $TYK_GW_PATH/gateway/version.go)
+
+# Plugin compiler arguments:
+#
+# - 1. plugin_name = vendor-plugin.so
+# - 2. plugin_id = optional, sets build folder to `/opt/plugin_{plugin_name}{plugin_id}`
+# - 3. GOOS = optional override of GOOS
+# - 4. GOARCH = optional override of GOARCH
+#
+# The script will build a plugin named according to the following:
+#
+# - `{plugin_name%.*}_{GATEWAY_VERSION}_{GOOS}_{GOARCH}.so`
+#
+# If GOOS and GOARCH are not set, it will build `{plugin_name}`.
+#
+# Example command: ./build.sh 
+# Example output: tyk-extras_5.0.0_linux_amd64.so
 
-CURRENTVERS=$(perl -n -e'/v(\d+).(\d+).(\d+)/'' && print "v$1\.$2\.$3"' $TYK_GW_PATH/gateway/version.go)
 plugin_name=$1
 plugin_id=$2
-# GOOS and GOARCH can be send to override the name of the plugin
-GOOS=$3
-GOARCH=$4
-CGOENABLED=0
+GOOS=${3:-$(go env GOOS)}
```

Comment:
Do we really need to explicitly manage `GOOS` and `GOARCH`?

The plugin compiler has the cross compiler toolchain to build supported `GOOS` and `GOARCH` so that this one image can build `arm64` plugins by adding `-e GOARCH=arm64` to the run command line. Even plugins having C dependencies should compile, in theory. So I see no need to explicitly manage these vars which are meant to be set by the user and transparently passed to `go build`.

Reply:
It's a good DX question for a new issue. The way the cross builds are made is by passing the GOOS and GOARCH as arguments, while we do pass other values as environment variables below, and I was tempted to make the breaking change for consistency.

Reply:
Filed under: https://tyktech.atlassian.net/browse/TT-8745